tailnet/telemetry.go

```diff
@@ -50,23 +61,65 @@ func (b *TelemetryStore) newEvent() *proto.TelemetryEvent {
 	defer b.mu.Unlock()
 
 	return &proto.TelemetryEvent{
-		Time:           timestamppb.Now(),
-		DerpMap:        DERPMapToProto(b.cleanDerpMap),
-		LatestNetcheck: b.cleanNetCheck,
+		Time:            timestamppb.Now(),
+		DerpMap:         DERPMapToProto(b.cleanDerpMap),
+		LatestNetcheck:  b.cleanNetCheck,
+		NodeIdSelf:      b.nodeIDSelf,
+		NodeIdRemote:    b.nodeIDRemote,
+		HomeDerp:        b.homeDerp,
+		ConnectionSetup: b.connSetupTime,
+		Application:     b.application,
 
 		// TODO(ethanndickson):
-		ConnectionAge:   &durationpb.Duration{},
-		ConnectionSetup: &durationpb.Duration{},
-		P2PSetup:        &durationpb.Duration{},
+		P2PSetup: &durationpb.Duration{},
 	}
 }
 
+func (b *TelemetryStore) markConnected(ip *netip.Addr, connCreatedAt time.Time, application string) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	b.connSetupTime = durationpb.New(time.Since(connCreatedAt))
+	b.connectedIP = ip
+	b.application = application
```

Comment:
This doesn't work for telemetry on coderd which can have many peers

Reply:
We'll focus on just the CLI telemetry for now.